common/nodes/database-schema.xml

```diff
@@ -380,42 +363,38 @@ CREATE TABLE storage_partition (
 	FOREIGN KEY (scope_map_id) REFERENCES scope_map(id) ON DELETE CASCADE
 );
 
-<!-- Miscellaneous -->
-
+<!-- Public Keys -->
 DROP TABLE IF EXISTS public_keys;
 CREATE TABLE public_keys (
- ID		int(11) NOT NULL auto_increment,
- Node		int(11) NOT NULL references nodes,
- Public_Key	varchar(4096) default NULL,
- PRIMARY KEY (ID)
+	id		INT AUTO_INCREMENT PRIMARY KEY,
+	node		INT NOT NULL,
+	public_key	VARCHAR(4096) NOT NULL,
```

Comment:
Interesting that this defaulted to NULL before. Was anything relying on that behavior?

I see `NOT NULL` added in lots of places throughout. Should that warrant a `BREAKING CHANGE` in the commit message? I'm just wondering about knock on effects from the subtle difference between a row with a  NULL in the column and a row with a column containing an empty string.

Reply:
The `public_keys` table isn't currently used, and it doesn't make any since to have a null for the public_key field, so I changed it.

Lots of places in the DB, we were setting default values, but as far as I can tell, we never actually rely on the default value anywhere in the code. It was probably a case of someone 15 years ago thought you needed to supply a default value when creating tables, which you don't (if you are always supplying field values in the SQL).